src/DataStructures/Variables.hpp

```diff
@@ -817,6 +829,17 @@ bool operator!=(const Variables<TagsList>& lhs,
   return not(lhs == rhs);
 }
 
+/// \ingroup DataStructuresGroup
+/// Construct a variables from the `Tensor`s in a `TaggedTuple`.
+template <typename... Tags>
+Variables<tmpl::list<Tags...>> variables_from_tagged_tuple(
```

Comment:
Can this be made a constructor for `Variables`?

Reply:
Ok not sure what you mean, I was thinking to make it a constructor since we also have one that constructs from another prefixed Variables, for instance. This function looks like a good candidate for a constructor to me. Your decision.

Reply:
The purpose of this function is to do template deduction for the class template.  Constructors can't do that without [deduction guides](https://en.cppreference.com/w/cpp/language/class_template_argument_deduction).  If we were on C++17 then this would make sense as a constructor.